via-router/src/router.rs

```diff
@@ -45,21 +45,21 @@ fn match_next_segment<'a, T>(
     segment: &str,
     range: [usize; 2],
 ) -> Option<Binding<'a, T>> {
-    let mut results: Option<SmallVec<_>> = None;
```

Comment:
While this seems like a reasonable optimization, I prefer a length check to determine whether or not we should return `Some` or `None`.

In a vacuum, there is no reason why that null pointer optimization would take away from the resilience of the router. However, if you factor in entropy, it's probably best if we do a len check.

Since `SmallVec` checks whether or not the inline storage spilled to the heap, it's slightly more complicated than checking a len ptr and as a result–it makes it more difficult to "trick" the router into thinking that the binding is empty.